pkg/plugin/util/util_test.go

```diff
@@ -101,4 +101,60 @@ var _ = Describe("Cover plugin util helpers", func() {
 			Expect(lines).To(Equal([]string{"noemptylines"}))
 		})
 	})
+
+	Describe("HasFileContentWith", Ordered, func() {
+		const (
+			path    = "testdata/PROJECT"
+			content = `# Code generated by tool. DO NOT EDIT.
+# This file is used to track the info used to scaffold your project
+# and allow the plugins properly work.
+# More info: https://book.kubebuilder.io/reference/project-config.html
+domain: example.org
+layout:
+- go.kubebuilder.io/v4
+- helm.kubebuilder.io/v1-alpha
+plugins:
+  helm.kubebuilder.io/v1-alpha: {}
+repo: github.com/example/repo
+version: "3"
+`
+		)
+
+		BeforeAll(func() {
+			err := os.MkdirAll("testdata", 0o755)
```

Comment:
We should not use the testdata dir at all.
We should do as in the other tests:

https://github.com/kubernetes-sigs/kubebuilder/blob/2dc541943ee5285b2bc4176338d8d64ff4e2f83c/test/e2e/deployimage/plugin_cluster_test.go#L38-L44

By using the TestContext we create a tmp dir for the tests and we ensure that we are using the kubebuilder binary build with the PR / source code changes